Reformating is_variable method

hirwa-nshuti opened the pull request:
This PR resolves #1799 

Added Docstrings examples to the `is_variable` method and also the array instance method with the same function.

hirwa-nshuti commented:
Hello, @sreegandh804 I want a PR review and as for the NumPy backend, the `is_variable` method always returns `False` 
I was suggesting if I can explicitly set the backends in implementing the examples. As it can clearly demonstrate the working of the method.
Thanks

sreegandh804 commented:
Hi @hirwa-nshuti ,
I had a couple of changes i would suggest:
1. I would suggest you correct the typos in the docstrings, like  'W' should be in uppercase and 'input' instead of 'inputs', 'Examples' instead of 'Functional Examples' etc... feel free to refer to the docs https://lets-unify.ai/ivy/contributing/4_open_tasks.html .
2. is_variable should be called as a function
3. Also, the examples should be With one ivy.Container input and With multiple ivy.Container inputs such that in the second case, exclusive would also be a container.

hirwa-nshuti commented:
Hi @sreegandh804 I have made the changes requested and my tests are passing this one failing is from other files.
Thanks

sreegandh804 commented:
Hi @hirwa-nshuti , it looks like you have deleted your implementation in ivy/array/gradients.py and the typos like 'W' should be in uppercase persist, please look into that. 🙂

hirwa-nshuti commented:
Hi @sreegandh804 
Sorry, I didn't notice that I removed the implementation in ivy/array/gradients.py, here I added it and also made the typos fixed.
Thanks

sreegandh804 commented on ivy/container/gradients.py:
```diff
@@ -16,6 +16,34 @@ def static_is_variable(
+
+        Examples
+        --------
+        with :code:`ivy.container` input:
```
'W' should be in uppercase and 'Container' instead of 'container'

sreegandh804 commented on ivy/container/gradients.py:
```diff
@@ -16,6 +16,34 @@ def static_is_variable(
+            b: false
+        }
+
+        with multiple :code:`ivy.Container` inputs:
```
'W' should be in uppercase and 'Container' instead of 'container'

sreegandh804 replied:
'W' should be in uppercase

sreegandh804 commented on ivy/container/gradients.py:
```diff
@@ -34,6 +62,34 @@ def is_variable(
+
+        Examples
+        --------
+        with :code:`ivy.Container` input:
```
'W' should be in uppercase

sreegandh804 commented:
Hi @hirwa-nshuti , I just noticed you opened multiple PRs, which is great that you are willing to contribute to our repo, but we would encourage you to open 1 PR at a time and open a new one once the old one has been merged/closed 😄 . Feel free to look though this page on our website https://lets-unify.ai/ivy/contributing/4_open_tasks.html.

hirwa-nshuti commented:
> Hi @hirwa-nshuti , I just noticed you opened multiple PRs, which is great that you are willing to contribute to our repo, but we would encourage you to open 1 PR at a time and open a new one once the old one has been merged/closed smile . Feel free to look though this page on our website https://lets-unify.ai/ivy/contributing/4_open_tasks.html.

Thank you 🤗 I will follow the pattern for sure.
